src/NHibernate/Impl/AbstractQueryImpl.cs

```diff
@@ -886,7 +886,7 @@ public IEnumerable<T> Future<T>()
 		{
 			if (!session.Factory.ConnectionProvider.Driver.SupportsMultipleQueries)
 			{
-				return List<T>();
+				return new DelayedEnumerator<T>(List<T>);
```

Comment:
Latent bug fix in `Future` method. In case of lack of multi-query supports, a raw list was yielded instead of the future list internal implementation.
But Linq Future handling casts results to the internal interface `IDelayedValue` for defining their post execute transformer. Previous code was doing that only if such a transformer was defined, and they are currently defined only on aggregating results using the `FutureValue` method, which does not have that flaw.

This also adjusts a "detail": previous implementation was causing an immediate execution of the query in case of lack of multiple query support, contrary to what do `FutureValue`. Now it will be delayed too, so a discarded future query will not have its query run at all when multiple queries are not supported, it will be run only if the future query is actually enumerated.